packages/flutter_driver/lib/src/driver.dart

```diff
@@ -205,6 +208,61 @@ class FlutterDriver {
     return result.text;
   }
 
+  /// Starts recording performance traces.
+  Future<Null> startTracing() async {
+    try {
+      await _appIsolate.invokeExtension(_kStartTracingMethod, const {});
+      return null;
+    } catch(error, stackTrace) {
+      throw new DriverError(
+        'Failed to start tracing due to remote error',
+        error,
+        stackTrace
+      );
+    }
+  }
+
+  /// Stops recording performance traces and downloads the trace profile.
+  Future<Map> stopTracingAndDownloadProfile() async {
+    Map<String, dynamic> stopResult =
+        await _appIsolate.invokeExtension(_kStopTracingMethod, const {});
+    String traceFilePath = stopResult['trace_file_path'];
+
+    // Tracing data isn't available immediately as some of it is queued up in
+    // the event loop.
+    Stopwatch sw = new Stopwatch()..start();
+    while(sw.elapsed < const Duration(seconds: 30)) {
+      Map<String, dynamic> downloadResult =
+          await _appIsolate.invokeExtension(_kDownloadTraceDataMethod, {
+            'trace_file_path': traceFilePath,
+          });
+
+      if (downloadResult['success'] == false) {
+        throw new DriverError('Failed to download trace file: $traceFilePath');
+      } else if (downloadResult['status'] != 'not ready') {
+        return downloadResult;
```

Comment:
Agreed. That's what all other commands do. The issue is that right now the structure of the trace is very much in flux. As of today we're downloading in chrome://tracing format. However, @chinmaygarde and @johnmccutchan are working a way to export our metrics into the Dart VM's timeline. So very soon we will be switching over to Dart service protocol. Once the dust settles I'll look into structurizing it.

I left a TODO to revisit once we have the new protocol to talk to.